src/domain/enrolment/__tests__/CreateEnrolmentPage.test.tsx

```diff
@@ -397,3 +397,60 @@ test('render and focuses error notification correctly', async () => {
 
   expect(screen.getByRole('textbox', { name: 'Nimi' })).toHaveFocus();
 });
+
+// FIXME: Tested in browser and the result is accepted. Unit test fails.
+test.skip('amount of adults is limited to 1 when there is 1 seat left after children', async () => {
+  render(<CreateEnrolmentPage />, {
+    mocks: mock3,
+    query: { eventId: eventId, occurrences: occurrenceIds },
+  });
+  await waitFor(() => {
+    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  });
+  userEvent.type(screen.getByLabelText(/lapsia/i), '19');
+  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
+  await act(wait);
+  expect(
+    screen.getByText(
+      /Arvon tulee olla enintään 1 yhdessä lasten lukumäärän kanssa/i
+    )
+  ).toBeInTheDocument();
+});
+
+// FIXME: Tested in browser and the result is accepted. Unit test fails.
+test.skip('amount of adults is limited to 0 when children fills the group in whole', async () => {
+  render(<CreateEnrolmentPage />, {
+    mocks: mock3,
+    query: { eventId: eventId, occurrences: occurrenceIds },
+  });
+  await waitFor(() => {
+    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  });
+  userEvent.type(screen.getByLabelText(/lapsia/i), '20');
+  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
+  await act(wait);
+  expect(
+    screen.getByText(
+      /Arvon tulee olla enintään 0 yhdessä lasten lukumäärän kanssa/i
+    )
+  ).toBeInTheDocument();
+});
```

Comment:
Couldn't get these to work. I don't even know what was wrong. They should mimic these: 
![image](https://user-images.githubusercontent.com/389204/107670527-801c4880-6c9b-11eb-95b9-f5a6cbddf10e.png)

Reply:
Testing can be tricky at first. I will check tomorrow

Reply:
Issue with the tests here is that the input where error should appears is not unfocused at any moment, which causes it to go to touched state, and then the error appears.

Can be fixed by just submitting the form or just using userEvent.tab() to focus out of the input.

```
  // userEvent.tab();

  userEvent.click(
    screen.getByRole('button', { name: 'Lähetä ilmoittautuminen' })
  );

  await waitFor(() => {
    expect(
      screen.getByText(
        /Arvon tulee olla enintään 1 yhdessä lasten lukumäärän kanssa/i
      )
    ).toBeInTheDocument();
  });
```

Reply:
worked!